src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -2,12 +2,121 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Diagnostics.Contracts;
+using System.Text;
 
 namespace System.IO
 {
     /// <summary>Contains internal path helpers that are shared between many projects.</summary>
     internal static partial class PathInternal
     {
+        internal const string ExtendedPathPrefix = @"\\?\";
```

Comment:
Note that I changed the terminology here to better reflect what it does. While it does allow long paths it also allows access to additional filenames and changes resolution behavior.